app/components/billing-status-badge.ts

```diff
@@ -22,10 +22,48 @@ export default class BillingStatusDisplayComponent extends Component<Signature>
     return this.currentUser?.activeDiscountForYearlyPlan || null;
   }
 
+  get activeDiscountForYearlyPlanExcludingStage2CompletionDiscount(): PromotionalDiscountModel | null {
+    return this.currentUser?.activeDiscountForYearlyPlanExcludingStage2CompletionDiscount || null;
+  }
+
+  get badgeType(): 'vip' | 'member' | 'old-discounts' | 'all-discounts' | 'free-weeks-left' | 'upgrade' {
```

Comment:
```suggestion
  get badgeType(): 'vip' | 'member' | 'discount-timer' | 'free-weeks-left' | 'upgrade' {
```

The template doesn't have to know whether what discount what we're rendering a discount timer for, no?